eureka-client/src/test/java/com/netflix/discovery/util/EurekaEntityFunctionsTests.java

```diff
@@ -0,0 +1,363 @@
+        applicationArrayList2.add(application2);
+        Applications applications2 = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList2);
+
+        Assert.assertEquals(2, EurekaEntityFunctions.mergeApplications(
+                applications1, applications2).size());
+    }
+
+    @Test
+    public void testMergeApplicationIfActionTypAddedReturnApplication() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo.getActionType())
+                .thenReturn(InstanceInfo.ActionType.ADDED);
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        Assert.assertEquals(application.getInstances(),
+                EurekaEntityFunctions.mergeApplication(
+                        application, application).getInstances());
+    }
+
+    @Test
+    public void testMergeApplicationIfActionTypModifiedReturnApplication() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo.getActionType())
+                .thenReturn(InstanceInfo.ActionType.MODIFIED);
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        Assert.assertEquals(application.getInstances(),
+                EurekaEntityFunctions.mergeApplication(
+                        application, application).getInstances());
+    }
+
+    @Test
+    public void testMergeApplicationIfActionTypDeletedReturnApplication() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo.getActionType())
+                .thenReturn(InstanceInfo.ActionType.DELETED);
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        Assert.assertNotEquals(application.getInstances(),
+                EurekaEntityFunctions.mergeApplication(
+                        application, application).getInstances());
+    }
+
+    @Test
+    public void testUpdateMetaIfNotNullReturnApplications() {
+        InstanceInfo instanceInfo = Mockito.mock(InstanceInfo.class);
+        Mockito.when(instanceInfo.getId()).thenReturn("foo");
+        Mockito.when(instanceInfo.getStatus())
+                .thenReturn(InstanceInfo.InstanceStatus.UP);
+        Application application = new Application("foo");
+        application.addInstance(instanceInfo);
+
+        ArrayList<Application> applicationArrayList = new ArrayList<>();
+        applicationArrayList.add(application);
+        Applications applications = new Applications(
+                "appsHashCode", 1559658285l, applicationArrayList);
+
+        Assert.assertEquals(applications.getRegisteredApplications(),
```

Comment:
Unsure what this assert achieves, do you want to check the version?

Reply:
changed